include/grpc/grpc_security.h

```diff
@@ -309,7 +326,8 @@ GRPCAPI void grpc_server_credentials_release(grpc_server_credentials *creds);
      NULL. */
 GRPCAPI grpc_server_credentials *grpc_ssl_server_credentials_create(
     const char *pem_root_certs, grpc_ssl_pem_key_cert_pair *pem_key_cert_pairs,
-    size_t num_key_cert_pairs, int force_client_auth, void *reserved);
+    size_t num_key_cert_pairs, int force_client_auth,
```

Comment:
@jboeuf is comment just for `credentials_create(` or also for `credentials_create_ex(`? (i guess i'm not privy to grpc's standard definition of legacy in this case.)

i suppose the solution is to keep the `void *reserved`? and the new options would be a new argument, which will break existing callers of this function (which Justin preferred not to).

Reply:
@jboeuf Could you say a bit more about this? Is your objection here of a technical or design nature?

If the former: the only thing any current caller can do is pass a `NULL`-valued `void *` since there's an assert check on the `NULL` value. After this change, all existing call sites can continue to pass a `NULL`-valued `void *` by way of implicit conversion.